### PR TITLE
Remove unnecesary library selection checking

### PR DESCRIPTION
**Pull request by wscho77**
If suffix is contained in the library name, there is no need to check suffix added name variation. 
Also, if delimiter is added in the library name, there is no need to check prefix added name variation.

I reorganized library selection code with below table.
| containsSuffix | containsDelim |  1st | 2nd | 3rd | 4th
| --- | --- | --- | --- | --- | --- |
| O | O | NameFmt | | | |
| O | X | NameFmt | PrefixNameFmt | | |
| X | O | NameSuffixFmt | NameFmt | | |
| X | X | PrefixNameSuffixFmt | NameSuffixFmt | PrefixNameFmt | NameFmt |

### CODE REVIEW

**Comment by jkotas** [comment]
https://github.com/dotnet/runtime/blob/47bcd1a8b053c2dcd05b47da033587199167e790/src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/LibraryNameVariation.Unix.cs#L18 has a managed copy of the algorithm that should be kept in sync

**Comment by jkotas** [comment]
Can this break loading of libraries with weird names? For example, liblib ?

**Comment by wscho77** [comment]
@jkotas, compare to the previous one, only adding another suffix case (even though suffix already exist) is removed.
So, liblib can be handled

**Comment by wscho77** [comment]
> https://github.com/dotnet/runtime/blob/47bcd1a8b053c2dcd05b47da033587199167e790/src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/LibraryNameVariation.Unix.cs#L18
> 
> has a managed copy of the algorithm that should be kept in sync

I also update [LibraryNameVariation.Unix.cs](https://github.com/dotnet/runtime/blob/47bcd1a8b053c2dcd05b47da033587199167e790/src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/LibraryNameVariation.Unix.cs#L18)

Line 18 in [47bcd1a](https://github.com/dotnet/runtime/commit/47bcd1a8b053c2dcd05b47da033587199167e790)

**Comment by jkotas** [comment]
> only adding another suffix case (even though suffix already exist) is removed.

Can this affect libraries with .so.so suffix then?

> only adding another suffix case (even though suffix already exist) is removed.

You have also changed the order for the last case.

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/LibraryNameVariation.Unix.cs [hunk 1]
```diff
@@ -35,31 +35,26 @@ internal static IEnumerable<LibraryNameVariation> DetermineLibraryNameVariations
 
                 bool containsDelim = libName.Contains(Path.DirectorySeparatorChar);
 
-                if (containsSuffix)
+                if (containsSuffix && containsDelim)
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/LibraryNameVariation.Unix.cs:38]: I am not sure whether the reformatting is an improvement. The new code is less compact and I think it makes it harder to see the differences between the different combinations.

**Comment by wscho77** [comment]
> > only adding another suffix case (even though suffix already exist) is removed.
> 
> Can this affect libraries with .so.so suffix then?
> 

Should we consider that kind of wiered case? I thought there was an intension not to add unnecessary variation because previous code checking suffix and delimiter.
If the goal is to consier ".so.so" and ".so.6.so" case, I think it is right to maintian the previous implementation.

> > only adding another suffix case (even though suffix already exist) is removed.
> 
> You have also changed the order for the last case.
> 

Yes. I thought it would be right to add Prefix first if there is no delimiter.

**Comment by AaronRobinsonMSFT** [comment]
@wscho77 I'm not sure This improves a lot and I prefer the current pattern. Is there a specific issue with the current logic?

/cc @elinor-fung

**Comment by wscho77** [comment]
@AaronRobinsonMSFT I just want to remove unnecessary check for finding library.
If the check for duplicate suffix case is intended, this patch is not necessary.
I close this PR.

**Pull request closed without merging**